### ISSUE TITLE
Submit-Renewal throws errors checking ARI on certs with no AKI

### ISSUE THREAD

**Issue by jamesfreeman959** [rmbolger/Posh-ACME#561]
Hi there,

I have recently started using Posh-ACME on Windows Server and really appreciate your efforts. Currently experiencing an issue on a machine I've recently installed it on. The machine already had Posh-ACME 3.9.0 installed from a trial run I made a few years ago but abandoned due to lack of time. I'm since getting things going again, and so upgraded Posh-ACME to 4.24.0 and then requested a new certificate using `New-PACertificate` - this worked on the first try. However I'm now having problems with both `Get-PACertificate` and `Submit-Renewal` - output from the latter is given below. I suspect this might be from my prior attempt at using Posh-ACME, but I'm not clear what I need to clear out or where to resolve the issue. Plus although the certificate I obtained doesn't need renewing, the 404 error buried in the output makes me think the renewal will fail.

Any help would be appreciated:

```
C:\> submit-renewal
Exception calling "GetInstance" with "1" argument(s): "unknown object in factory: System.Byte[]
Parameter name: obj"
At C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\4.24.0\Public\Get-PACertificate.ps1:71 char:13
+             $akiBytes = [Org.BouncyCastle.Asn1.X509.AuthorityKeyIdent ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

ConvertTo-Base64Url : Cannot bind argument to parameter 'Bytes' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\4.24.0\Public\Get-PACertificate.ps1:73 char:56
+ ...          $ariID = '{0}.{1}' -f (ConvertTo-Base64Url $akiBytes),(Conve ...
+                                                         ~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [ConvertTo-Base64Url], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,ConvertTo-Base64Url

WARNING: ARI request failed.
Update-PAOrder : {
  "type": "urn:ietf:params:acme:error:malformed",
  "detail": "Must specify a request path",
  "status": 404
}
At C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\4.24.0\Public\Submit-Renewal.ps1:47 char:21
+                     Update-PAOrder -Order $order
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Update-PAOrder], WebExcept
   ion
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Update-PAOrder
WARNING: Order 'redacted.example.com' is not recommended for renewal yet. Use -Force to override.
```

**Comment by rmbolger** [comment]
Hi @jamesfreeman959, thanks for reaching out. I don't think you did anything wrong here. I have a feeling I made a bad assumption in the new ARI code in that all certs contained an AKI extension. So there's a point where it needs to read that value to generate the ARI ID used to reference that cert and if the cert doesn't have it, it currently breaks.

I'm wondering if Let's Encrypt didn't always include that value in their certs and the really old cert it's trying to check is causing the bug. You can probably verify by just running `Get-PACertificate` all by itself against that problem order and it should generate the same error.

A short term fix would likely be to delete your existing config (or just the specific order folder) and start from scratch. You can find the account root location by running `(Get-PAAccount).Folder`. However, before you delete the problem config, can you find the `cert.cer` file in that folder, open it in notepad, and post the contents here? That's the public part of the cert and would help me a lot having a real cert causing the issue to test against.

**Comment by rmbolger** [comment]
I may not need that problem cert after all. I found an old test cert that caused the same problem and there's now a fix in the code that will go out with the next version.

**Comment by rmbolger** [comment]
The fix is now live in 4.25.0
